core/src/main/scala/com/banno/kafka/consumer/ConsumerOps.scala

```diff
@@ -345,13 +347,32 @@ case class ConsumerOps[F[_], K, V](consumer: ConsumerApi[F, K, V]) {
     * failed records when the consumer is closed, and will likely result in less
     * reprocessing after a failure. The consumer must be configured to disable
     * offset auto-commits.
+    *
+    * If keepAliveInterval is provided and finite, and topics have been manually
+    * assigned, then committed offsets will be periodically refreshed to prevent
+    * them from expiring for low traffic topics.
     */
-  def readProcessCommit[A](pollTimeout: FiniteDuration)(
+  def readProcessCommit[A](
+      pollTimeout: FiniteDuration,
+      keepAliveInterval: Duration = Duration.Inf,
+  )(
       process: ConsumerRecord[K, V] => F[A]
-  )(implicit F: ApplicativeError[F, Throwable]): Stream[F, A] =
-    consumer
-      .recordStream(pollTimeout)
-      .evalMap(r => process(r) <* consumer.commitSync(r.nextOffset))
+  )(implicit F: Temporal[F]): Stream[F, A] =
```

Comment:
Just to double check: this breaks compatibility, but we do not care because the next release of `kafka4s` is bound to be a major bump?

Reply:
Yup.

The bincompat change is a real pain because we would have a mix of overloads and default arguments which are a very uneasy combination ... much simpler to do the obvious thing and accept the breakage.